### ISSUE TITLE
Poor Scrolling Performance

### ISSUE THREAD

**Issue by kyledecot** [kirbyt/KTPhotoBrowser#2]
When displaying a photo browser w/ around 100 images, the scrolling becomes very jagged and sometimes the view freezes for a few seconds. The images are being pulled in from the web using the SDWebImage classes.

**Comment by kyledecot** [comment]
Also should mention that I am using the latest iPhone 4.0 beta

**Comment by kirbyt** [comment]
I haven't tried 4.0 yet, but I'll look into it.

**Comment by kyledecot** [comment]
It seems that when using 4.0 whenever you start to scroll, layout subviews gets called repeatedly which is causing the slow performance. I fixed it by basically renaming the layoutSubviews method to layoutPhotos and then setting up a notification for when the orientation changes and calling layoutPhotos from there.

**Comment by kirbyt** [comment]
Sorry for the long delay. The thumbnail scrolling performance has been greatly improved. I added a new sample project that shows scrolling with 1000 images, and the scrolling is fast and smooth.

My apologies for taking so long to resolve this issue.

**Comment by kyledecot** [comment]
This problem still seems to be prevalent. I'm displaying 25 images that are using the KTThumbView+SDWebImage category.

**Comment by kirbyt** [comment]
Hi @itsfall,

Sorry for the delayed response.

Make sure you have the most recent code. I made a major change a few weeks ago to improve the performance of thumbnail scrolling. Also, today I added more Flickr images to the SDWebImage sample for a total of 50 images. I tested the code on different devices including a 1st generation iPod touch, and I do not see the jagged scrolling problem.

One thing to keep in mind is that you want to keep your thumbnail images small, ideally the same size as the thumbnail display size. The default is 75x75. Loading a full size image to display as a thumbnail will slow down scrolling because it has to scale down the image as the user scrolls. This can be slow. The LocalImageSample shows how to pre-render a 75x75 thumbnail after selecting an image from the camera or photo library. The SDWebImage sample downloads two versions of the image, a large size (approximately 640x480) and a thumbnail size (75x75).

Hope that helps.

-KIRBY
